allennlp/commands/find_learning_rate.py

```diff
@@ -0,0 +1,286 @@
+                             start_lr: float,
+                             end_lr: bool,
+                             num_batches: int,
+                             linear_steps: bool,
+                             stopping_factor: Optional[float]) -> None:
+    """
+    Runs learning rate search for given `num_batches` and saves the results in ``serialization_dir``
+
+    Parameters
+    ----------
+    trainer: :class:`~allennlp.common.registrable.Registrable`
+    params : ``Params``
+        A parameter object specifying an AllenNLP Experiment.
+    serialization_dir : ``str``
+        The directory in which to save results.
+    start_lr: ``float``
+        Learning rate to start the search.
+    end_lr: ``float``
+        Learning rate upto which search is done.
+    num_batches: ``int``
+        Number of mini-batches to run Learning rate finder.
+    linear_steps: ``bool``
+        Increase learning rate linearly if False exponentially.
+    stopping_factor: ``float``
+        Stop the search when the current loss exceeds the best loss recorded by
+        multiple of stopping factor. If ``None`` search proceeds till the ``end_lr``
+    """
+
+    prepare_environment(params)
+
+    os.makedirs(serialization_dir, exist_ok=True)
+
+    check_for_gpu(params.get('trainer').get('cuda_device', -1))
+
+    all_datasets = datasets_from_params(params)
+    datasets_for_vocab_creation = set(params.pop("datasets_for_vocab_creation", all_datasets))
+
+    for dataset in datasets_for_vocab_creation:
+        if dataset not in all_datasets:
+            raise ConfigurationError(f"invalid 'dataset_for_vocab_creation' {dataset}")
+
+    logger.info("From dataset instances, %s will be considered for vocabulary creation.",
+                ", ".join(datasets_for_vocab_creation))
+    vocab = Vocabulary.from_params(
+            params.pop("vocabulary", {}),
+            (instance for key, dataset in all_datasets.items()
+             for instance in dataset
+             if key in datasets_for_vocab_creation)
+    )
+
+    model = Model.from_params(vocab=vocab, params=params.pop('model'))
+    iterator = DataIterator.from_params(params.pop("iterator"))
+    iterator.index_with(vocab)
+
+    train_data = all_datasets['train']
+
+    trainer_params = params.pop("trainer")
+    no_grad_regexes = trainer_params.pop("no_grad", ())
+    for name, parameter in model.named_parameters():
+        if any(re.search(regex, name) for regex in no_grad_regexes):
+            parameter.requires_grad_(False)
+
+    trainer = Trainer.from_params(model,
+                                  serialization_dir,
+                                  iterator,
+                                  train_data,
+                                  params=trainer_params,
+                                  validation_data=None,
+                                  validation_iterator=None)
+
+    learning_rates, losses = search_learning_rate(trainer, start_lr,
```

Comment:
Could you add a couple of logging statements here telling the user what is happening - e.g `Starting Learning Rate Search` and `Saving Learning rate plot to {some path}`

Reply:
Will add it and the docs and ping